src/SourceGenerators/Uno.UI.Tasks/Content/Uno.UI.Tasks.targets

```diff
@@ -43,18 +43,46 @@
 		<PropertyGroup>
 			<XamarinProjectType Condition="'$(UnoForceProcessPRIResource)'!=''"></XamarinProjectType>
 		</PropertyGroup>
-		<ResourcesGenerationTask_v0 Resources="@(PRIResource)"
+
+		<ItemGroup>
+			<_UnoTempPRIResource Include="@(PRIResource)" />
+		</ItemGroup>
+
+		<!-- Assign the Link property so files defined in shared projects preserve their natural path structure -->
+		<AssignLinkMetadata Items="@(PRIResource)"
+							Condition="'@(PRIResource)' != '' and '%(PRIResource.DefiningProjectFullPath)' != '$(MSBuildProjectFullPath)'">
+			<Output TaskParameter="OutputItems" ItemName="_Temp" />
+		</AssignLinkMetadata>
+
+		<ItemGroup>
+			<!-- remove new linked items so their new metadata is used instead -->
+			<_UnoTempPRIResource Remove="@(_Temp)" />
+			<_UnoTempPRIResource Include="@(_Temp)" />
+
+			<!-- Clear temporary group -->
+			<_Temp Remove="@(_Temp)" />
+		</ItemGroup>
```

Comment:
Would it be possible to Output to an undeclared `_UnoTempPRIResource` to avoid doing gymnastics ?

Then we could have a conditional ItemGroup that includes `<_UnoTempPRIResource Include="@(PRIResource)" />` if `AssignLinkMetadata` didn't run !